src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,21 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> letters = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while (reader.ready()) {
+                int character = reader.read();
+                if (Character.isLetter(character)) {
+                    letters.add((char) Character.toLowerCase(character));
+                }
+            }
+        }
+
+        StringBuilder result = new StringBuilder();
+        for (Character c : letters) {
+            result.append(c);
+        }
+        return result.length() < 5 ? result.toString() : result.toString().substring(0, 5);
```

Comment:
make 5 a constant field
and consider checking in a loop if you have already exceeded the maximum size